examples/03_Multitask_Exact_GPs/Hadamard_Multitask_GP_Regression.ipynb

```diff
@@ -108,7 +123,7 @@
     "        \n",
     "        return gpytorch.distributions.MultivariateNormal(mean_x, covar)\n",
     "\n",
-    "likelihood = gpytorch.likelihoods.GaussianLikelihood()\n",
+    "likelihood = gpytorch.likelihoods.HadamardGaussianLikelihood(num_tasks=2)\n",
```

Comment:
Rather than changing the tutorial to only use task-specific noises, it would be good to instead show both cases and discuss the pros and cons of both approaches in the tutorial. Ideally you could produce the output you shared in https://github.com/pytorch/botorch/issues/2765#issuecomment-2715182824

We could also try to highlight a potential failure mode where in the very low data regime andwith uniform noise levels across tasks, estimating task-specific noises results in worse performance due to the less parsimonious model.